actions/actions.py

```diff
@@ -41,9 +49,54 @@ def resume(args):
     resume_unit_helper(ConfigRenderer(CONFIG_FILES))
 
 
+def vhost_queue_info(vhost):
+    try:
+        output = subprocess.check_output([RABBITMQ_CTL,
+                                          '-p', vhost, 'list_queues',
+                                          'name', 'messages', 'consumers'])
+        queue_info = []
+        if '...done' in output:
+            queues = output.split('\n')[1:-2]
+        else:
+            queues = output.split('\n')[1:-1]
+
+        for queue in queues:
+            [qname, qmsgs, qconsumers] = queue.split()
+            queue_info.append({
+                'name': qname,
+                'messages': int(qmsgs),
+                'consumers': int(qconsumers)
+            })
+
+        return queue_info
+
+    except Exception as ex:
+        # if no queues, just raises an exception
+        log(str(ex), level='DEBUG')
+        return []
+
+
+def list_unconsumed_queues(args):
+    """Resume the Ceilometer services.
+    @raises Exception should the service fail to start."""
+    count = 0
+    for vhost in list_vhosts():
+        iterator = 0
+        for queue in vhost_queue_info(vhost):
+            if queue['consumers'] == 0:
+                vhostqueue = "unconsumed-queues.{}.{}".format(vhost, iterator)
+                action_set({vhostqueue: "{} - {}".format(queue['name'],
+                                                         queue['messages'])})
+                iterator = iterator + 1
+                count = count + 1
+
+    action_set({'unconsumed-queue-count': count})
+
```

Comment:
You might consider action_fail'ing on exceptions.  Like: https://github.com/openstack/charm-percona-cluster/blob/master/actions/actions.py#L73